### ISSUE TITLE
Finish the implementation and integration of multiclassing.

### ISSUE THREAD

**Issue by farm-zaq** [uchicago-cs/chiventure#910]
Finish the implementation and integration of mutliclassing. This will allow players and/or game devs to combine classes, adding many more options for character customization. 
1) This necessarily includes finishing the basic implementation, which would entail combining the effects, attributes, and skills, and skill inventories of both class. These should be implemented as helper functions (like the existing multiclass_shortdesc) which can be called upon by the multiclass function, particularly in lines with TODO comments. This also includes finding and updating outside calls to class information that may be modified by changes to the class structure. These are necessary for multiclassing to properly function.
2) Update documentation (mainly of relevant readme pages, but also of relevant wiki pages) so that changes are actually usable for devs.
3) Expand class tests to test multiclassing (this would also include researching how testing currently works in chiventure). 
4) If any important concerns are brought up in the currently open PR, these should be addressed as well.
5) Time permitting, other minor changes could be made to improve functionality, such as improved formatting for string combinations, updating data structures, addressing minor concerns brought up in the PR, addressing redundancies between the two classes, and addressing potential overflow issues in string concatenations. These are all issues that don't stop multiclassing from functioning or only cause issues in incredibly niche circumstances, so they are of lesser priority.

**Comment by farm-zaq** [comment]
I've only been updating by updating the wiki and PR description, so I'll summarize those updates here (and will make more updates in the future). Since this issue was created, I have implemented functionality to combine attributes, skill trees, and skill inventories. I have also updated outside calls to class-> name to look at class-> name and all parent class names, to make sure implementation is sensical (so, if a knight can't use a weapon, a multiclass that is part knight can't, either, for example).

**Comment by farm-zaq** [comment]
I've added functionality to combine the effects for the classes.
When checking my code, I've realized that the combination of attributes currently does not work at all (it actually has been assigning an int to the relevant pointer, and it currently would interfere with the attributes of the base class, which is undesirable).

**Comment by farm-zaq** [comment]
The bugs for effects have been sorted out. Because it is an isolated addition. (and, likely, my last addition for this sprint), I have just added it to the old branch and rerequested some review in the PR.
Major implementation TODO items: attributes
Other important TODO items: readme, testing

**Comment by farm-zaq** [comment]
After two weeks, the remains incomplete, meaning that another issue will have to be opened in a future sprint to finish this. None of the tasks really have been finished, but I will remark on the progress and unfinished aspects of each one.
1) Every element of the basic implementation has been created, other than a helper function to combine the attributes of the two classes. Currently, the multiclass function sets the attributes to NULL. Because of this, multiclassing is technically incomplete; however, due to the fact that attributes are currently called upon nowhere in all of chiventure, it is technically still a functioning module that could be integrated.
2) No updates to the documentation have been made. The wiki page needs to be cleaned, and nothing has been done for the readmes.
3) No testing has been created.
4) No concerns were raised in the PR, so this could be considered a success. With that said, the PR hasn't finished being reviewed, so this task should find itself on future multiclassing issues because of that.
5) None of the minor concerns listed in the issue description or on the wiki page have been implemented.
In summary, the multiclassing module is in a state where it can be integrated into chiventure, but it should be be documented, tested, and cleaned before being considered finished.

**Comment by firatciftci** [comment]
Issue Score: ✔️++

Comments:
Great job on this! The updated that you have provided give a really detailed overview of your work. The closing statement also makes it clear the progression and future of the issue at hand.
